docs/docs/top-level-api.md

```diff
@@ -40,3 +40,61 @@ export const Router = createRouter({
   }),
 });
 ```
+
```

Comment:
I would've kept it separated, so the the order remains:

- top-level to create a router
- details of the router
- lower level stuff

so that we're in line with the "recommended" usage: using router first, and location only if needed.

Reply:
Ok, but `location` didn't make sens to me, since encodeSearch / decodeSearch is not really tied to the location object + useLocation is but is in hooks.
Not sure of how we can name this page.

Reply:
Or "Lower level API"? And keep a header that say "sometimes you need to perform more complex stuff bla bla"

Reply:
Yup, Lower level API sounds good to me

Reply:
All done ✅